src/utils/FileDownloader.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from "fs-extra";
+import * as path from "path";
+
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
+import { Readable } from "stream";
+import { v4 as uuid } from "uuid";
+import { ExtensionContext, Uri, window } from "vscode";
+import { localize } from '../localize';
+
+export class FileDownloader {
+
+    public getDownloadStoragePath(context: ExtensionContext): string {
+        return path.join(context.globalStorageUri.fsPath, `file-downloader-downloads`);
+    }
+
+    public getDownloadedFile(fileName: string, context: ExtensionContext): Uri {
+        const downloadStoragePath = this.getDownloadStoragePath(context);
+        return Uri.file(path.join(downloadStoragePath, fileName));
+    }
+
+    public async downloadExternalBinary(url: string, fileName: string, context: ExtensionContext): Promise<Uri> {
+
+        const options: AxiosRequestConfig = {
+            timeout: 20000,
+            responseType: `stream`,
+            headers: {
+                Accept: "application/octet-stream",
+                "Content-Type": "application/octet-stream"
```

Comment:
Content-Type header usually needed for requests with body, probably we don't need it here

Reply:
Thanks, tested this and you are correct! 👍